jina/peapods/runtimes/gateway/prefetch.py

```diff
@@ -16,6 +17,21 @@
 class PrefetchMixin(ABC):
     """JinaRPCServicer """
 
+    def __init__(self):
+        self._message_buffer: Dict[str, Future[Message]] = dict()
+        asyncio.create_task(self._receive())
```

Comment:
I do think this `_receive` should be scheduled in the `fetch_to.append()` call

Reply:
We need to run the receive coroutine permanently in the background I think as we don't know when our responses will arrive.
One should probably also add some timeout mechanism to recover from disappearing messages, but this is not part of the PR at the moment (as it did not exist before as well).